File: 1-easy/problems/2869-minimum-operations-to-collect-elements/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def minOperations(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        lookup = [False]*k
        for i in reversed(range(len(nums))):
            if nums[i] > len(lookup) or lookup[nums[i]-1]:
                continue
            lookup[nums[i]-1] = True
            k -= 1
            if not k:
                break
        return len(nums)-i

File: 1-easy/problems/2869-minimum-operations-to-collect-elements/solution.ts
function minOperations(nums: any, k: any): boolean | number | string | any {
  """
  :type nums: List[int]
  :type k: int
  :rtype: int
  """
  lookup = [false]*k
  for (i in reversed(Array.from({length: (nums}, (_, i) => i).length))) {
      if (nums[i] > (lookup).length || lookup[nums[i]-1]) {
          continue
      lookup[nums[i]-1] = true
      k -= 1
      if (not k) {
          break
  return (nums).length-i;
}

export { minOperations };
